stdlib/bytes.ml

```diff
@@ -314,6 +314,70 @@ let rcontains_from s i c =
   else
     try ignore (rindex_rec s i c); true with Not_found -> false
 
+let filter p arg =
+  let size_arg = length arg in
```

Comment:
Isn't it possible to share part of the logic with `Array`?
(At least the conversion of eight booleans into a char.)

Reply:
But where would the function go? You'd be adding a dependency between basic library modules.

Reply:
If we cannot (or don't want to) share part of the logic,
we should probably add comments to indicate that any
modification should be replicated in the other module.